### ISSUE TITLE
Wallet Connect - Pasting a URI and clicking on the 'How to copy the dApp URI' link while loading results in the connection getting stuck

### ISSUE THREAD

**Issue by virginiabalducci** [status-im/status-desktop#16887]
# Bug Report

## Steps to reproduce

1. click on the dapp connections 
2. select Wallet Connect
3. Paste a URI and while the loading icon displays click on 'How to copy the dApp URI' link

#### Expected behavior
The connection to be established without issues

#### Actual behavior
The connection does not establish, the spinner keeps on loading and nothing happens

https://github.com/user-attachments/assets/05d02ca8-e943-4fbe-ab57-b841cd1886c6

Pasting a fresh URI (2) shows an error message

https://github.com/user-attachments/assets/fe44731e-3f26-4730-af1e-f0025f36fd6c

Pasting the same URI (2) again establishes the connection

https://github.com/user-attachments/assets/70b2c65d-9570-438e-9941-7cd410177fd0

Console errors:

[app_20241204_183333.log](https://github.com/user-attachments/files/18024492/app_20241204_183333.log)

```
ERR 2024-12-05 14:07:14.336Z qt error                                   topics="qt" tid=115998701 text="[object Object] Missing or invalid. Record was recently deleted - proposal: 1733407322198651" file=qrc:/app/AppLayouts/Wallet/services/dapps/sdk/generated/bundle.js:2 category=js
DBG 2024-12-05 14:07:14.341Z qt message                                 topics="qt" tid=115998701 text="WC WalletConnectSDK.onProposalExpire; details: {\"id\":1733407322198651}" file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:491 category=qml
ERR 2024-12-05 14:08:01.021Z qt error                                   topics="qt" tid=115998701 text="[object Object] Missing or invalid. Record was recently deleted - proposal: 1733407370997097" file=qrc:/app/AppLayouts/Wallet/services/dapps/sdk/generated/bundle.js:2 category=js
DBG 2024-12-05 14:08:01.025Z qt message                                 topics="qt" tid=115998701 text="WC WalletConnectSDK.onProposalExpire; details: {\"id\":1733407370997097}" file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:491 category=qml
INFO [12-05|14:08:06.621] New local node record                    seq=1,733,348,017,314 id=b48733c74de5c947 ip=192.168.0.92 udp=49859 tcp=44069
INFO [12-05|14:08:06.623] New local node record                    seq=1,733,348,017,315 id=b48733c74de5c947 ip=192.168.0.92 udp=49859 tcp=44069
INFO [12-05|14:08:12.274] New local node record                    seq=1,733,348,017,316 id=b48733c74de5c947 ip=186.127.96.97 udp=49859 tcp=44069
ERR 2024-12-05 14:08:31.131Z qt error                                   topics="qt" tid=115998701 text="[object Object] Missing or invalid. Record was recently deleted - proposal: 1733407396013722" file=qrc:/app/AppLayouts/Wallet/services/dapps/sdk/generated/bundle.js:2 category=js
DBG 2024-12-05 14:08:31.132Z qt message                                 topics="qt" tid=115998701 text="WC WalletConnectSDK.onProposalExpire; details: {\"id\":1733407396013722}" file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:491 category=qml
INFO [12-05|14:09:16.604] New local node record                    seq=1,733,348,017,317 id=b48733c74de5c947 ip=186.127.96.97 udp=49859 tcp=44069
INFO [12-05|14:09:17.601] New local node record                    seq=1,733,348,017,318 id=b48733c74de5c947 ip=186.127.96.97 udp=49859 tcp=44069
ERR 2024-12-05 14:10:20.935Z qt error                                   topics="qt" tid=115998701 text="[object Object] Missing or invalid. Record was recently deleted - proposal: 1733407483809745" file=qrc:/app/AppLayouts/Wallet/services/dapps/sdk/generated/bundle.js:2 category=js
```

### Additional Information

- Status desktop version: 2.31 RC1
- Operating System: Mac OS Sonoma

Note: This seems like an edge case, not sure if it's worth adding it to 2.32 work? @alexjba what do you think?

**Comment by virginiabalducci** [comment]
This also happens when user clicks on the X button while the spinner is loading

https://github.com/user-attachments/assets/8a2fd30c-7173-4b6b-8f42-629c3eb97ddc

**Comment by alexjba** [comment]
> Note: This seems like an edge case, not sure if it's worth adding it to 2.32 work? @alexjba what do you think?

I think we're missing something in the connect flow. Can't really explain the behavior. Let's add it for 2.32

**Comment by virginiabalducci** [comment]
This was fixed on #16893 16893
